packages/next-on-pages/src/buildApplication/processVercelFunctions/ast.ts

```diff
@@ -2,58 +2,72 @@ import type { Node } from 'acorn';
 import type * as AST from 'ast-types/gen/kinds';
 import type { ProcessVercelFunctionsOpts } from '.';
 import { cliError } from '../../cli';
+import { createHash } from 'crypto';
 
 /**
  * Collects the identifiers from the AST and adds them to the provided maps and lists.
  *
  * @param collectedIdentifiers The collected identifiers.
- * @param program The AST for the function file.
+ * @param programInfo The AST for the function file and its entrypoint.
  * @param opts The options for the function processing.
  */
 export function collectIdentifiers(
 	{ identifierMaps, programIdentifiers }: CollectIdentifiersOpts,
-	program: AST.ProgramKind,
+	programInfo: ProgramInfo,
 	{ disableChunksDedup }: Partial<ProcessVercelFunctionsOpts>,
 ): void {
 	// Wasm
 	collectIdentifiersByType(
 		{ programIdentifiers, foundIdentifiers: identifierMaps.wasm },
-		program,
+		programInfo,
 		extractWasm,
 	);
 
 	// Manifests
 	collectIdentifiersByType(
 		{ programIdentifiers, foundIdentifiers: identifierMaps.manifest },
-		program,
+		programInfo,
 		extractManifest,
 	);
 
 	// Webpack chunks
 	if (!disableChunksDedup) {
 		collectIdentifiersByType(
 			{ programIdentifiers, foundIdentifiers: identifierMaps.webpack },
-			program,
+			programInfo,
 			extractWebpack,
 		);
 	}
 }
 
+/**
+ * Groups the identifiers in the provided maps by their consumers.
+ *
+ * @param identifierMaps The maps of found identifiers in the AST.
+ */
+export function groupIdentifiers(identifierMaps: IdentifierMaps) {
+	// Webpack Chunks
+	groupIdentifiersByType(identifierMaps.webpack);
+	// Manifests
+	groupIdentifiersByType(identifierMaps.manifest);
```

Comment:
are you missing wasm? or are you intentionally skipping it?

if it needs to be included, what about doing:
```suggestion
	Object.values(identifierMaps).forEach(map => groupIdentifiersByType(map));
```

Reply:
Wasm is just imported wasm files that are moved, so there is no point to include it here since grouping is used for code blocks.

Reply:
sorry I am not sure I get it, are you saying that wasm files don't get grouped?

if not, it might not be too beneficial but I don't see a negative in treating them as the other type of identifiers and grouping them, no?

Reply:
Well, it's just that it's completely pointless extra work to do. Only code blocks get grouped and written to grouped files. Including Wasm in the grouping logic is just making it do extra work at build time for no gain.

Wasm deduping is deduping the assets that are imported, not deduping code blocks.

Reply:
yeah.. as I said that grouping wasms probably wouldn't be very beneficial, but also the extra work would be negligible, wouldn't it?

Feels to me that it would make sense to treat all identifier types consistently, slightly clearer code, very small benefit in the output, very small extra work done at build time

But it's not like it's going to make a huge difference so if you're totally against it I'm fine keeping it this way, but at least I would imagine that we should document somewhere that wasms are not grouped (to avoid possible future confusions) (as far as I can tell that is not clearly indicated anywhere in the code, is it?)

Reply:
i will concede in this instance when it comes to that 99% of cases will have no wasm identifiers so there won't really be anything running